YoutubeExplode/Bridge/PlaylistExtractor.cs

```diff
@@ -106,60 +141,70 @@ internal partial class PlaylistExtractor
             .EnumerateArrayOrNull()?
             .Select(j => new ThumbnailExtractor(j))
             .ToArray() ??
+        
+        TryGetSidebarPrimary()?
+            .GetPropertyOrNull("thumbnailRenderer")?
+            .GetPropertyOrNull("playlistCustomThumbnailRenderer")?
+            .GetPropertyOrNull("thumbnail")?
+            .GetPropertyOrNull("thumbnails")?
+            .EnumerateArrayOrNull()?
+            .Select(j => new ThumbnailExtractor(j))
+            .ToArray() ??
 
         Array.Empty<ThumbnailExtractor>()
     );
 
-    private JsonElement? TryGetContentRoot() => Memo.Cache(this, () =>
-        // Initial response
-        _content
-            .GetPropertyOrNull("contents")?
-            .GetPropertyOrNull("twoColumnBrowseResultsRenderer")?
-            .GetPropertyOrNull("tabs")?
-            .EnumerateArrayOrNull()?
-            .ElementAtOrNull(0)?
-            .GetPropertyOrNull("tabRenderer")?
-            .GetPropertyOrNull("content")?
-            .GetPropertyOrNull("sectionListRenderer")?
-            .GetPropertyOrNull("contents")?
-            .EnumerateArrayOrNull()?
-            .ElementAtOrNull(0)?
-            .GetPropertyOrNull("itemSectionRenderer")?
-            .GetPropertyOrNull("contents")?
-            .EnumerateArrayOrNull()?
-            .ElementAtOrNull(0)?
-            .GetPropertyOrNull("playlistVideoListRenderer")?
-            .GetPropertyOrNull("contents") ??
-
-        // Continuation response
-        _content
-            .GetPropertyOrNull("onResponseReceivedActions")?
-            .EnumerateArrayOrNull()?
-            .ElementAtOrNull(0)?
-            .GetPropertyOrNull("appendContinuationItemsAction")?
-            .GetPropertyOrNull("continuationItems")
+    public string? TryGetPlayListUrl() => Memo.Cache(this, () =>
+     TryGetPlaylistProperty()?
+         .GetPropertyOrNull("playlistShareUrl")?
+         .GetStringOrNull()
     );
 
     public IReadOnlyList<PlaylistVideoExtractor> GetVideos() => Memo.Cache(this, () =>
-        TryGetContentRoot()?
+        TryGetPlaylistProperty()?
+            .GetPropertyOrNull("contents")?
             .EnumerateArrayOrNull()?
-            .Select(j => j.GetPropertyOrNull("playlistVideoRenderer"))
+            .Select(j => j.GetPropertyOrNull("playlistPanelVideoRenderer"))
             .WhereNotNull()
             .Select(j => new PlaylistVideoExtractor(j))
             .ToArray() ??
 
         Array.Empty<PlaylistVideoExtractor>()
     );
 
+    public string? TryGetLastVideoId() => Memo.Cache(this, () =>
+        GetVideos()?
+            .LastOrDefault()?
+            .TryGetVideoId()
+    );
+
+    public int? TryGetLastIndex() => Memo.Cache(this, () =>
+        TryGetPlaylistProperty()?
+            .GetPropertyOrNull("contents")?
+            .EnumerateArrayOrNull()?
+            .Select(j => j.GetPropertyOrNull("playlistPanelVideoRenderer"))
```

Comment:
This could probably be replaced with `GetVideos` too?

Reply:
Yes, just have to add a method that to ```PlaylistVideoExtractor``` that gets the video index.

Reply:
Yeah, let's do that